espresso/core/java/androidx/test/espresso/matcher/ViewMatchers.java

```diff
@@ -757,7 +759,9 @@ private WithTextMatcher(Matcher<String> stringMatcher) {
     @Override
     protected void describeMoreTo(Description description) {
       description.appendText("view.getText() with or without transformation to match: ");
-      stringMatcher.describeTo(description);
+      StringDescription buffer = new StringDescription();
```

Comment:
This will make anywhere that requires adapting special chars verbose to implement. You could have a private helper class here which takes in a Matcher<String> matcher, and then attach its description. Something like:
`specialCharDescribeTo(stringMatcher, description)`.

Reply:
See my other comment, it doesn't look like `appendHighlightedValue` is a good way to hide the difference between handling a value v.s. a text. In fact, `SpecialCharVisualizer` is only capable of working with strings, right?